# t-Student
Distribución y estadístico que aproxmiman a la normal cuando no se conoce el desvío de la muestra
###### https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.t.html#scipy.stats.t
###### For the noncentral t distribution, see nct.
## Pesos de bebes al nacer - Efectos de bebés que consumieron cocaína

In [1]:
import numpy as np
from scipy.stats import t
import matplotlib.pyplot as plt

### Pesos de bebes al nacer
#### Pesos de bebés de madres que consumen cocaína
+ En un estudio de los efectos sobre los bebés del consumo de cocaína durante el embarazo, se obtuvieron los siguientes datos de pesos al nacer:
    + n=190, X̅=2700 g, s=645 g 
    + (según datos de “Cognitive Outcomes of Preschool Children with Prenatal Cocaine Exposure”, 
       + de Singer et al., Journal of the American Medical Association, vol. 291, núm. 20). 
    + El diseño del estudio justifica el supuesto de que la muestra puede tratarse como una muestra aleatoria simple. 
+ Utilice los datos muestrales para construir un estimado del intervalo de confianza del 95% de µ, 
    + el peso medio al nacer de todos los bebés hijos de madres que consumieron cocaína durante el embarazo.

In [2]:
# Primero verificar los requisitos de t-Student
# a. Muestra aleatoria simple: cada posible muestra del mismo tamaño n tiene la misma posibilidad de ser elegida.
# b. n = 190. => n > 30. Aunque el problema no dice que la población está normalmente distribuida. n > 30 y se cumple el TCL.
# c. No hay datos de 𝞂, pero si de s.
n = 190
x = 2700
df = n -1
s = 645
se = s /n**0.5
confidence_level = 0.95

# Hay una función concreta para esto que es .interval()
print(f"Los datos dados y calculados son:")
print(f"Tamaño muestra: {n}, Grados de Libertad: {df}, Desvío Std: {s}, Error Std:{se}")
tmin, tmax = t.interval(confidence_level, df, loc=x, scale=se)
print(f"Resultado: Intevalo de confianza {tmin:.0f} < µ < {tmax:.0f}")

Los datos dados y calculados son:
Tamaño muestra: 190, Grados de Libertad: 189, Desvío Std: 645, Error Std:46.79321813209575
Resultado: Intevalo de confianza 2608 < µ < 2792


**Interpretación**
+ Con base en los datos muestrales, tenemos una confianza del 95% de que los límites de 2608 g y 2792 g realmente contienen el valor del peso medio al nacer. 
+ Ahora podemos comparar este resultado con un intervalo de confianza construido para los pesos al nacer de niños cuyas madres no consumieron cocaína.
+   (a continuación...).

### Pesos de bebés al nacer
#### Pesos de bebés de madres que NO consumen cocaína
+ Una muestra aleatoria de los pesos al nacer de 186 bebés tiene una media de 3103 g y una desviación estándar de 696 g 
    + (según datos de “Cognitive Outcomes of Preschool Children with Prenatal Cocaine Exposure”, 
        + de Singer et al., Journal of the American Medical Association, vol. 291, núm. 20). 
+ Estos bebés son hijos de madres que no consumieron cocaína durante el embarazo. 
+ Construya un estimado del intervalo de confianza del 95% del peso medio al nacer de todos estos bebés. 

In [3]:
# Primero verificar los requisitos de t-Student
# a. Muestra aleatoria simple: cada posible muestra del mismo tamaño n tiene la misma posibilidad de ser elegida.
# b. n = 186. => n > 30. Aunque el problema no dice que la población está normalmente distribuida. n > 30 y se cumple el TCL.
# c. No hay datos de 𝞂, pero si de s.
n=186
df = n-1
x = 3103
s=696
se = s / n ** 0.5
confidence_level = 0.95
# Hay una función concreta para esto que es .interval()
print(f"Los datos dados y calculados son:")
print(f"Tamaño muestra: {n}, Grados de Libertad: {df}, Desvío Std: {s}, Error Std:{se}")
tmin, tmax = t.interval(confidence_level, df, loc=x, scale=se)
print(f"Resultado: Intevalo de confianza {tmin:.0f} < µ < {tmax:.0f}")

Los datos dados y calculados son:
Tamaño muestra: 186, Grados de Libertad: 185, Desvío Std: 696, Error Std:51.03319602743095
Resultado: Intevalo de confianza 3002 < µ < 3204


## Comparación de los resultados
+ Se compara el resultado con el intervalo de confianza obtenido en la muestra de bebés de madres que no consumieron cocaína,
    + con los pesos al nacer de hijos de madres que consumieron cocaína durante el embarazo. 
+ **Conclusiones**
+ Los intervalos de confianza no están ni siquiera solapados
    + son claramente mayores los peros de los bebés hijos de madres que no consumieron.
+ Al parecer si existen diferencias y estas diferencias sugieren que:
    + Los bebés hijos de madres que no consumen cocaína son más pesados
    + Los hijos de madres que sí consumen cocaína nacen con menor peso

___________________________________________________________________________________________________________________________________________________________